### DISCUSSION TITLE
Hide category from the home index page

### DISCUSSION THREAD

**Discussion by ghost** [mediacms-io/mediacms#708] [Q&A] [answered]
Is it possible to exclude a specific category from the home page? I have some videos cluttering the home page and it would look much cleaner if I could exclude that category. First impressions are the most important, right? ☺️

**Comment by mgogoulos** [comment, marked as answer]
Hi, just sharing some thoughts here

The home page displays three sections, 'featured', 'latest', and 'recommended', but you can choose to hide any of the three. In order to make any change to one of these three sections, you have to touch the code, and add an exclude parameter (eg don't show videos of X category, or show videos only for Y category). The featured section can also achieve this, you have to explicitly mark a media as 'globally featured', so that it appears there, and thus you can hide the other sections (latest, recommended) and only keep featured, and keep control of what's there. 

Another useful feature is implemented in https://cinemata.org/ and is the ability to make a playlist and show it on the index page, as you can see below 'Recent Videos' there is a number of items that is totally controlled by the admin of the portal, so you could add multiple playlists and show them on the index page. This is top on my list to bring to the open source MediaCMS version but is something I haven't worked yet. 

Hope this help!

**Reply by ghost** [reply]
Thanks for the ideas! If one were to try and add the exclude parameter, which file(s) would need to be changed? I’ll go the Featured media route if I’m unable to figure it out, which is very likely. 😅

**Reply by mgogoulos** [reply]
Check this example, under the Django shell:

```
Media.objects.filter().count()                                                                                                                                                                                                                                                   
769

Media.objects.filter().exclude(category__title='Documentary').count()                                                                                                                                                                                                            
650
```

So you may find the queries that power all of the listings the index page is using under file files.views.py MediaList.get

instead of 

```
            if show_param == "featured":
                media = Media.objects.filter(basic_query, featured=True)
            else:
                media = Media.objects.filter(basic_query).order_by("-add_date")
```

what you want is 

```
            if show_param == "featured":
                media = Media.objects.filter(basic_query, featured=True)
            else:
                media = Media.objects.filter(basic_query).exclude(category__title__in=['Documentary', 'Art', 'whatever']).order_by("-add_date")
```

for the latest media.

Two notes here: 

1. The code that does this calculation (eg get latest media) can be improved, and get to be read more intuitively than now (there are just too many if/else conditions there that make reading/changing/maintaining it hard
2. keep in mind that adding any field as include or exclude on a queryset results in increase of time spent on the DB to fetch queries. This will not have impact if you have a small number of media (eg < 10k) but can have an impact if you have a big number for media

**Reply by ghost** [reply]
This worked perfectly, thank you so much! I'll monitor db performance over time even though I doubt we'll ever get close to 10k media.
